indexer/services/comlink/src/controllers/api/v4/orders-controller.ts

```diff
@@ -129,17 +128,24 @@ class OrdersController extends Controller {
         return OrderTable.orderIdToUuid(redisOrder.order!.orderId!);
       },
     );
+    const postgresOrderIdsToFetch: string[] = _.difference(
+      redisOrderIds,
+      _.map(postgresOrders, OrderColumns.id),
+    );
 
-    // Fetch additional orders from Postgres that are not in the initial query, because they
-    // might not conform to the status filter above. For example, if the user is querying for
-    // `status: [BEST_EFFORT_OPENED]`, we need to fetch all orders from Postgres, because if the
-    // order in postgres is 'OPENED', then we do not want to return this order to the user as
-    // 'BEST_EFFORT_OPENED'.
-    const additionalPostgresOrders: OrderFromDatabase[] = await OrderTable.findAll({
-      id: redisOrderIds,
-    }, [], {
-      ...DEFAULT_POSTGRES_OPTIONS,
-    });
+    // Postgres is regarded as the source of truth, so for any redis orders not returned from the
+    // initial postgres query, we need to fetch them from Postgres to ensure we have the most
+    // accurate status. For example, if the user is querying for `status: [BEST_EFFORT_OPENED]`,
+    // we need to fetch all orders from Postgres, because if the order in postgres is 'OPENED',
+    // then we do not want to return this order to the user as 'BEST_EFFORT_OPENED'.
+    let additionalPostgresOrders: OrderFromDatabase[] = [];
+    if (!_.isEmpty(postgresOrderIdsToFetch)) {
+      additionalPostgresOrders = await OrderTable.findAll({
```

Comment:
nit: does it make sense to filter on `statuses` here? Just because there could be a race condition where some order exists in the redis cache, but by the time you go to get it from the database, it's in some unexpected status like `CANCELED`

Reply:
I think in this case, it would be correct to use the status found in postgres for the order, e.g. if the user was looking for all open / best-effort opened orders and by the time we fetch from postgres, the order is canceled, then we shouldn't return the order.